staging/src/k8s.io/apimachinery/pkg/util/validation/field/errors.go

```diff
@@ -49,14 +48,10 @@ func (v *Error) ErrorBody() string {
 	case ErrorTypeRequired, ErrorTypeForbidden, ErrorTypeTooLong, ErrorTypeInternal:
 		s = fmt.Sprintf("%s", v.Type)
 	default:
-		var bad string
-		badBytes, err := json.Marshal(v.BadValue)
-		if err != nil {
-			bad = err.Error()
-		} else {
-			bad = string(badBytes)
-		}
-		s = fmt.Sprintf("%s: %s", v.Type, bad)
+		// TODO: internal types have panic guards against json.Marshalling to prevent
+		// accidental use of internal types in external serialized form.  For now, use
+		// %#v, although it would be better to show a more expressive output in the future
+		s = fmt.Sprintf("%s: %#v", v.Type, v.BadValue)
```

Comment:
I tried it, it puts newlines in, displays type and length info (i.e. it prints `(string) (len=4) "foo"`) which ends up being not helpful in the other direction.